src/versionbits.cpp

```diff
@@ -105,6 +106,47 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     return state;
 }
 
+// return the numerical statistics of blocks signalling the specified BIP9 condition in this current period
+UniValue AbstractThresholdConditionChecker::GetStateStatisticsFor(const CBlockIndex* pindexPrev, const Consensus::Params& params, ThresholdConditionCache& cache) const
+{
+    UniValue obj(UniValue::VOBJ);
+    
+    int elapsedBlocks;
+    int nPeriod = Period(params);
+    int nThreshold = Threshold(params);
+    
+    const CBlockIndex* currentIndex = pindexPrev;
+    
+    obj.push_back(Pair("period length", nPeriod));
+    obj.push_back(Pair("threshold", nThreshold));
+
+    // Find beginning of period
+    if (pindexPrev != NULL) {
+        pindexPrev = pindexPrev->GetAncestor(pindexPrev->nHeight - ((pindexPrev->nHeight + 1) % nPeriod));
+        elapsedBlocks = currentIndex->nHeight - pindexPrev->nHeight;
+        obj.push_back(Pair("elapsed", elapsedBlocks));
+    }
+    
+    // Walk back from current block to beginning of period
+    std::vector<const CBlockIndex*> vToCompute;
+    while (pindexPrev->nHeight != currentIndex->nHeight){
```

Comment:
What if `pindexPrev == currentIndex == NULL`? Also, why not keep a rolling count rather than using an accessory `vector`?